test/system/src/main/java/io/pravega/test/system/framework/services/PravegaProperties.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2019 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system.framework.services;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class encapsulates all system properties to be passed to the operator for a Pravega deployment.
+ * */
+@Slf4j
+@Builder
+public class PravegaProperties {
+    private static final ImmutableMap<String, String> DEFAULT_PROPERTIES = ImmutableMap.<String, String>builder()
+            // Segment store properties
+            .put("autoScale.muteInSeconds", "120")
+            .put("autoScale.cooldownInSeconds", "120")
+            .put("autoScale.cacheExpiryInSeconds", "120")
+            .put("autoScale.cacheCleanUpInSeconds", "120")
+            .put("curator-default-session-timeout", "10000")
+            .put("bookkeeper.bkAckQuorumSize", "3")
+            // Controller properties.
+            .put("controller.transaction.maxLeaseValue", "60000")
+            .put("controller.retention.frequencyMinutes", "2")
+            .put("log.level", "DEBUG")
+            .build();
+
+    private final Map<String, String> properties = new HashMap<String, String>(DEFAULT_PROPERTIES);
+    private PravegaProperties(){}
+
+    /**
+     * This method adds a key only if it is absent, assuming the default values never need to change
+     *
+     * */
+    private void addProperty(String key, String value) {
+        properties.putIfAbsent(key, value);
+    }
+
+    public ImmutableMap<String, String> getProperties() {
+        return ImmutableMap.copyOf(properties);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder strBuilder = new StringBuilder();
+        for (String e: this.getProperties().keySet()) {
+            strBuilder.append(e);
+            strBuilder.append(":");
+            strBuilder.append(this.getProperties().get(e));
+            strBuilder.append("|");
+        }
+        return strBuilder.toString();
+    }
+
+    /**
+     * This class overrides the lombok builder.
+     * Add all methods for building PravegaProperties (with/without Auth/TLS) to this class
+     */
+    public static final class PravegaPropertiesBuilder {
+        private static final PravegaProperties DEFAULT_INSTANCE = new PravegaProperties();
```

Comment:
I'm a bit confused about this declaration, it feels like `builder().build()` will always return the same instance, while `builder().buildWithSecurity()` returns a new instance each time. Can you elaborate on the behavior you are after with this `DEFAULT_INSTANCE`?

Reply:
DEFAULT_INSTANCE has properties that will result in a pravega cluster being created without security (security disabled).
buildWithSecurity() returns an instance with all properties including those required for security enablement.
Both are methods return immutable instance of PravegaProperties.  
buildWithSecurity() could be optimized to return a static instance instead of creating a new instance every time.

Still better would be to read all properties from a properties file in the resources folder. There could be multiple resource files, one for each deployment flavour (no security, with auth, with tls+auth) and each builder would read the appropriate resource file.

Reply:
> DEFAULT_INSTANCE has properties that will result in a pravega cluster being created without security (security disabled).

perhaps for the needs of our current tests, this is sufficient. it doesn't seem to be flexible, though. the `PravegaProperties` instance is static and final, which does not enable any customization. I agree that changing properties via code is not ideal, and config files might work, but I wonder if we won't have the need of configuring Pravega separately for different test cases.

to sum up, it seems to work fine for the current state of affairs, but it might be limiting for our future use, and I'd like to know your view since you have been working on this code and thinking about it. what should we do?

Reply:
I have removed the 'PravegaProperties' class and added a method to Utils class to read properties from the appropriate properties file(s) under resources folder.